Sources/FigmaExportCore/Processor/AssetsProcessor.swift

```diff
@@ -90,184 +90,155 @@ public struct ImagesProcessor: AssetsProcessable {
+        let pairs = makeSortedAssetPairs(lightSet: lightSet, lightHCSet: lightHCSet)
+        return .success(pairs, warning: warning)
     }
 
-    private func validateAndMakePairs(light: [AssetType], dark: [AssetType]) -> ProcessingPairResult {
-
+    private func validateAndMakePairs(light: [AssetType],
+                                      dark: [AssetType],
+                                      lightHC: [AssetType],
+                                      darkHC: [AssetType]) -> ProcessingPairResult {
         // Error checks
-
         var errors = ErrorGroup()
-
-        // 1. countMismatch
+        // CountMismatch
         if light.count < dark.count {
             errors.all.append(AssetsValidatorError.countMismatch(light: light.count, dark: dark.count))
         }
-
-        // 2. foundDuplicate
-        var lightSet: Set<AssetType> = []
-        light.forEach { asset in
-
-            // badName
-            if !isNameValid(asset.name) {
-                errors.all.append(AssetsValidatorError.badName(name: asset.name))
-            }
-
-            switch lightSet.insert(asset) {
-            case (true, _):
-                break // ok
-            case (false, let oldMember): // already exists
-                errors.all.append(AssetsValidatorError.foundDuplicate(assetName: oldMember.name))
-            }
-        }
-
-        var darkSet: Set<AssetType> = []
-        dark.forEach { asset in
-            switch darkSet.insert(asset) {
-            case (true, _):
-                break // ok
-            case (false, let oldMember): // already exists
-                errors.all.append(AssetsValidatorError.foundDuplicate(assetName: oldMember.name))
-            }
+        if light.count < lightHC.count {
+            errors.all.append(AssetsValidatorError.countMismatchLightHighContrastColors(light: light.count, lightHC: lightHC.count))
         }
-
-        // 3. darkAssetNotFoundInLightPalette
-        let darkElements = darkSet.subtracting(lightSet)
-        if !darkElements.isEmpty {
-            errors.all.append(AssetsValidatorError.darkAssetsNotFoundInLightPalette(assets: darkElements.map { $0.name }))
+        if dark.count < darkHC.count {
+            errors.all.append(AssetsValidatorError.countMismatchDarkHighContrastColors(dark: dark.count, darkHC: darkHC.count))
         }
-
-        // 4. descriptionMismatch
+        // FoundDuplicate
+        let lightSet: Set<AssetType> = foundDuplicate(assets: light, errors: &errors, isLightAssetSet: true)
+        let darkSet: Set<AssetType> = foundDuplicate(assets: dark, errors: &errors)
+        let lightHCSet: Set<AssetType> = foundDuplicate(assets: lightHC, errors: &errors)
+        let darkHCSet: Set<AssetType> = foundDuplicate(assets: darkHC, errors: &errors)
+        // AssetNotFoundInLightPalette
+        checkSubtracting(firstAsset: lightSet, secondAsset: darkSet, errors: &errors)
+        checkSubtracting(firstAsset: lightSet, secondAsset: lightHCSet, errors: &errors)
```

Comment:
Method `checkSubtracting` can generate `darkAssetsNotFoundInLightPalette` error only but you specified light and light high contrast colors.